impl/spanner/testutil/testutil.go

```diff
@@ -58,33 +58,27 @@ func inMemClient(ctx context.Context, t testing.TB, dbName string) (*spanner.Cli
 	if err != nil {
 		t.Fatalf("Starting in-memory fake: %v", err)
 	}
+	t.Cleanup(srv.Close)
 	srv.SetLogger(t.Logf)
 	dialCtx, cancel := context.WithTimeout(ctx, 1*time.Second)
 	defer cancel()
 	conn, err := grpc.DialContext(dialCtx, srv.Addr, grpc.WithInsecure())
 	if err != nil {
-		srv.Close()
 		t.Fatalf("Dialing in-memory fake: %v", err)
 	}
+	t.Cleanup(func() { conn.Close() })
```

Comment:
Can you drop the wrapper function here? i.e. `t.Cleanup(conn.Close)` ?

Reply:
`conn.Close` returns `error` which doesn't quite match the function signature for `Cleanup`